### PR TITLE
Write out QSIRecon pipeline-wise reports

### PR DESCRIPTION
**Pull request by tsalo**
Closes none. Partially reimplements https://github.com/PennLINC/qsiprep/pull/761.

Probably doesn't work yet.

## Changes proposed in this pull request

- Add nireports as a dependency and use it.
- Add QSIRecon report configs.
- Add data loader module based on fMRIPrep.

### CODE REVIEW

**Comment by tsalo** [comment]
@mattcieslak this is ready to review. None of the HTML reports have anything useful except for the boilerplate, but we can deal with that in a future PR.

File: docs/api.rst
Comment (mattcieslak) [inline docs/api.rst:1]: I was planning on deleting this entirely.

File: qsirecon/cli/run.py [hunk 1]
```diff
@@ -171,39 +171,53 @@ def main():
...
+        qsirecon_suffixes = []
+        for node_spec in workflow_spec["nodes"]:
+            qsirecon_suffix = node_spec.get("qsirecon_suffix", None)
+            qsirecon_suffixes += [qsirecon_suffix] if qsirecon_suffix else []
+
+        qsirecon_suffixes = sorted(list(set(qsirecon_suffixes)))
+        config.loggers.cli.info(f"QSIRecon pipeline suffixes: {qsirecon_suffixes}")
+        failed_reports = []
+        for qsirecon_suffix in qsirecon_suffixes:
+            suffix_dir = str(config.execution.output_dir / f"qsirecon-{qsirecon_suffix}")
```
Comment (mattcieslak) [inline qsirecon/cli/run.py:200]: this is a smart approach

File: qsirecon/data/reports-spec-anat.yml
Comment (mattcieslak) [inline qsirecon/data/reports-spec-anat.yml:1]: This might not ever be used in a recon workflow

File: qsirecon/tests/data/amico_noddi_outputs.txt [hunk 2]
```diff
@@ -4,19 +4,20 @@ logs/CITATION.bib
...
 qsirecon-NODDI/sub-PNC/dwi/sub-PNC_acq-realistic_space-T1w_desc-preproc_model-NODDI_mfp-AMICOconfig_dwimap.pickle.gz
 qsirecon-NODDI/sub-PNC/dwi/sub-PNC_acq-realistic_space-T1w_desc-preproc_model-noddi_mdp-icvf_dwimap.nii.gz
 qsirecon-NODDI/sub-PNC/dwi/sub-PNC_acq-realistic_space-T1w_desc-preproc_model-noddi_mdp-isovf_dwimap.nii.gz
 qsirecon-NODDI/sub-PNC/dwi/sub-PNC_acq-realistic_space-T1w_desc-preproc_model-noddi_mdp-od_dwimap.nii.gz
 qsirecon-NODDI/sub-PNC/dwi/sub-PNC_acq-realistic_space-T1w_desc-preproc_model-noddi_mfp-direction_dwimap.nii.gz
-qsirecon/dwiqc.json
-qsirecon/sub-PNC
-qsirecon/sub-PNC
-qsirecon/sub-PNC.html
 sub-PNC
```
Comment (mattcieslak) [inline qsirecon/tests/data/amico_noddi_outputs.txt:23]: Do you know why these keep getting created?

**Review by mattcieslak — APPROVED** [review]
None of the comments should hold up merging this IMO

File: qsirecon/data/reports-spec-anat.yml
Comment (tsalo) [reply qsirecon/data/reports-spec-anat.yml:1]: We just need _something_ when there are enough sessions that we want to split the reports by session.

File: qsirecon/tests/data/amico_noddi_outputs.txt (hunk 2 above)
Comment (tsalo) [reply qsirecon/tests/data/amico_noddi_outputs.txt:23]: That's where the config TOML file and any crash files will be written

**Pull request merged**